### PR TITLE
feat: get Connectathon record details from API

### PR DESCRIPTION
**Pull request by whichwit**
* new API endpoint `/connectathon`
* dynamically display Connectathon record details
* use generic record index for saving test results and as default death certificate number
* once set, prefer state data over props data
* add unit tests

### CODE REVIEW

**Comment by whichwit** [comment]
PR not ready to merge, pending new library releases from https://github.com/nightingaleproject/vrdr-dotnet/pull/459

**Comment by johnarleyburns** [comment]
Maybe you can link canary against the master nuget and then we can update
it to the release nuget as part of the canary release process?

**Comment by whichwit** [comment]
Possibly. Let me see what options I have.

File: canary/Controllers/ConnectathoController.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+        /// <summary>
+        /// Returns all Connectathon test messages.
+        /// GET /connectathon
+        /// </summary>
+        [HttpGet("Connectathon")]
+        [HttpGet("Connectathon/Index")]
+        public DeathRecord[] GetTestConnectathonDeathRecords()
+        {
+            return Connectathon.Records;
```
Comment (johnarleyburns) [inline canary/Controllers/ConnectathoController.cs:27]: Please update this test with the recent changes.

**Review by johnarleyburns — CHANGES_REQUESTED** [review]

File: canary/Controllers/ConnectathoController.cs (hunk 1 above)
Comment (whichwit) [reply canary/Controllers/ConnectathoController.cs:27]: API unit test added

File: canary.tests/ConnectathonControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        {
+            // Arrange
+            var controller = new ConnectathonController();
+
+            // Act
+            var result = controller.Index();
+
+            // Assert
+            var deathRecordArray = Assert.IsType<DeathRecord[]>(result);
+            Assert.Equal(3, result.Length);
```
Comment (johnarleyburns) [inline canary.tests/ConnectathonControllerTests.cs:24]: Nice test

**Review by johnarleyburns — APPROVED** [review]

**Pull request merged**